acnportal/acnsim/events/event.py

```diff
@@ -120,53 +132,38 @@ def _from_dict(
         return out_obj, loaded_dict
 
 
-class UnplugEvent(Event):
+class PluginEvent(EVEvent):
+    """ Subclass of Event for EV plugins.
+
+    Args:
+        timestamp (int): See Event.
+        ev (EV): The EV which will be plugged in.
+    """
+
+    def __init__(self, timestamp: int, ev: "EV") -> None:
+        super().__init__(timestamp, ev)
+        self.event_type = "Plugin"
+        self.precedence = 10
+
+
+class UnplugEvent(EVEvent):
     """ Subclass of Event for EV unplugs.
 
     Args:
         timestamp (int): See Event.
-        station_id (str): ID of the EVSE where the EV is to be unplugged.
-        session_id (str): ID of the session which should be ended.
+        ev (EV): The EV which will be unplugged.
     """
 
-    def __init__(self, timestamp, station_id, session_id):
-        super().__init__(timestamp)
+    def __init__(self, timestamp: int, ev: "EV") -> None:
+        super().__init__(timestamp, ev)
         self.event_type = "Unplug"
-        self.station_id = station_id
-        self.session_id = session_id
```

Comment:
@zach401 just realized we should probably also have accessors for these in case anyone was using them.